### PR TITLE
Refactoring less and update dist filenames.

### PR DESCRIPTION
**Pull request by vsn4ik**
- <s>Bower not supported `ignore` option</s>
- <s>`user-select` without prefix not exists (http://caniuse.com/#feat=user-select-none)</s>
- Fixed Bootstrap 2 `.add-on` cursor and `.add-on i` position
- Fixed Bootstrap 3 `.input-group-addon` cursor
- Fixed "dist" filenames in `bower.json` and `composer.json`
- Fixed some `csslint` errors
- Rename `docs/REAME.md` to `docs/README.md`
- Add `clearDates` to docs
- Move `.jscsrc` to `js/.jscsrc`

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+    "dist/css/bootstrap-datepicker.css",
+    "dist/css/bootstrap-datepicker3.css",
+    "js/bootstrap-datepicker.js"
+  ],
   "license": "Apache-2.0",
   "dependencies": {
     "jquery" : ">=1.7.1",
     "bootstrap" : ">=2.0.4 <4.0"
-  },
-  "ignore": {}
```
Comment (acrobat) [inline bower.json]: I've added this empty ignore tag because of #1199, so this should be preserved to avoid error's when running bower install

**Comment by vsn4ik** [comment]
Thanks! Сorrected `ignore` to array.

File: less/datepicker3.less [hunk 2]
```diff
@@ -50,26 +50,21 @@
...
-	}
-	table{
+	table {
 		margin: 0;
 		-webkit-touch-callout: none;
 		-webkit-user-select: none;
 		-khtml-user-select: none;
 		-moz-user-select: none;
 		-ms-user-select: none;
-		user-select: none;
```
Comment (hebbet) [inline less/datepicker3.less]: Why did you remove that?

Comment (vsn4ik) [inline less/datepicker3.less]: This property without vendor prefix not exist (not supported by someone).

Comment (vsn4ik) [inline less/datepicker3.less]: https://developer.mozilla.org/en-US/docs/Web/CSS/user-select

**Comment by acrobat** [comment]
Looks good at first, will check whole PR tonight and then it should be ready to merge!

File: less/datepicker3.less (hunk 2 above)
Comment (hebbet) [inline less/datepicker3.less]: ok, bootstrap also adds user-input without prefixes

https://github.com/twbs/bootstrap/blob/master/less/mixins/vendor-prefixes.less#L222

Comment (acrobat) [inline less/datepicker3.less]: Maybe we should keep the property as it does no harm to the code. Csslint also throws a warning for it when the `user-select`property is not set.

```
WARNING: Missing standard property 'user-select' to go along with '-moz-user-select'. When using a vendor-prefixed property, make sure to include the standard one. (vendor-prefix) Browsers: All
```

so my idea is to keep the property

Comment (vsn4ik) [inline less/datepicker3.less]: Done.

**Comment by acrobat** [comment]
Thanks @vsn4ik!
